### ISSUE TITLE
Escape first of title if lowercase

### ISSUE THREAD

**Issue by bwiernik** [retorquere/zotero-better-bibtex#952]
If the first letter of the first word of a title is lowercase, it is probably reasonable to assume that this was intentional and that the word should not be uppercase under any circumstances. In such cases, I think that BBT should escape the first word or letter to prevent case changes.

For example, currently BBT exports this title:
psychmeta: an R package for psychometric meta-analysis

using Better BibLaTeX as:
```
@article{Dahlkepsychmetapackagepsychometric,
  langid = {english},
  title = {Psychmeta: An {{R}} Package for Psychometric Meta-Analysis},
  shorttitle = {Psychmeta},
  journaltitle = {Applied Psychological Measurement},
  shortjournal = {Appl. Psychol. Meas.},
  author = {Dahlke, Jeffrey A. and Wiernik, Brenton M.},
  pubstate = {in press}
}
```

rather than:
```
@article{Dahlkepsychmetapackagepsychometric,
  langid = {english},
  title = {{psychmeta}: An {{R}} Package for Psychometric Meta-Analysis},
  shorttitle = {{psychmeta}},
  journaltitle = {Applied Psychological Measurement},
  shortjournal = {Appl. Psychol. Meas.},
  author = {Dahlke, Jeffrey A. and Wiernik, Brenton M.},
  pubstate = {in press}
}
```

**Comment by retorquere** [comment]
@fbennett, I use the citeproc titlecaser to do this. Is this something you're interested in as a behavior? If not, I can work around it (I already mess around with the titlecaser results).

@njbart, general comments?

**Comment by fbennett** [comment]
It looks reasonable to me, to except the first word from uppercasing if its first letter is lowercase in the source. @njbart?

**Comment by retorquere** [comment]
Reasonable as in "I'll change the citeproc-js title caser"?

**Comment by retorquere** [comment]
I promise I won't rise up. I can vouch for my teenager son that he won't rise up either (not before noon at least).

**Comment by njbart** [comment]
I’d agree this might make sense, but I’m afraid it wouldn’t be in accordance with the CSL specs:

> **Title Case Conversion**
> Title case conversion (with text-case set to “title”) for English-language items is performed by:
> 1\.    For [all-]uppercase strings, the first character of each word remains capitalized. All other letters are lowercased.
> 2\.    For lower or mixed case strings, the first character of each lowercase word is capitalized. The case of words in mixed or uppercase stays the same.
> In both cases, stop words are lowercased, unless they are the first or last word in the string, or follow a colon. (http://docs.citationstyles.org/en/1.0.1/specification.html#title-case-conversion)

I’ve been using the – admittedly ugly, but established – `<span class="nocase">`…`</span>` syntax in such cases, which does get properly escaped upon export by BBT.

**Comment by retorquere** [comment]
Uh-oh, the uprising is here :worried: .

(JK, @njbart makes a good point, we don't want to violate spec here)

**Comment by retorquere** [comment]
Still, the point here is about preserving intent. If the first letter of a sentence is lowercased, in a sentence-cased field (as the Zotero fields are), then it makes sense for BBT to not uppercase this first letter. And possibly protect the first word, but that is going to be tricky.

**Comment by bwiernik** [comment]
It would certainly help me with citing R packages 😀

**Comment by retorquere** [comment]
Can you get me a sample reference by way of right-click  / BBT / error report?

**Comment by bwiernik** [comment]
ID YTT6483J

**Comment by retorquere** [comment]
@njbart the current fallout from this is [here](https://circleci.com/gh/retorquere/zotero-better-bibtex/7184).

**Comment by retorquere** [comment]
So I've run some tests, and MLA for example capitalizes `Psychmeta` in that reference unless you add `nocase`. That means that the intent in Zotero/citeproc is that all-lowercase is non-protected.

**Comment by bwiernik** [comment]
I guess I'm maybe not understanding the BibTeX convention of storing titles in Title Case? If I use citeproc-js with Zotero natively with a sentence case style (i.e., no case is specified) such as APA, the test reference I submitted is rendered correctly as `psychmeta: psychometric meta-analysis toolkit`. But if I try to use the same style with, for example pandoc, it outputs incorrectly as `Psychmeta: psychometric meta-analysis toolkit`, because BBT converted the title to Title Case in the interim.

According to Chicago and MLA styles, Psychmeta should be capitalized in this reference, but not in APA or other sentence case styles. Zotero/citeproc handles this by assuming that all titles are stored in sentence case, but BBT breaks this assumption by storing them all in Title Case.

**Comment by bwiernik** [comment]
Okay, looking at https://github.com/retorquere/zotero-better-bibtex/issues/383, apparently this is a (to me, completely bizarre) convention of bib(la)tex. Rather than this complicated workaround I suggested, could you just make the title-casing an optional preference so that BBT can be used with pandoc, which follows the CSL conventions?

**Comment by retorquere** [comment]
*shrug* bib(la)tex is pretty bizarre (this is one of the more mundane points really), but it predates citeproc by a wide margin. I assume there were good reasons to go with title-case as the base. I'm not sure why citeproc chose to go differently other than that sentence case appeals do a different aesthetic.

You can set the hidden preference `suppressTitleCase` to true, and BBT will back off.

**Comment by retorquere** [comment]
BTW, pandoc is just wrong on how they interpret Bib(La)TeX, and not trivially wrong either: they're going against behavior confirmed by the biblatex authors. People who want CSL semantics should use `Better CSL JSON` or `Better CSL YAML`, which avoids both the problem of the (by necessity) imperfect translation of Zotero references to BibTeX, followed by the broken parsing of these references of pandoc. This has come up time and again. This is a clear pandoc bug that in my opinion they should fix, or tell people about the pitfalls of using the broken BibTeX interpreter in pandoc.

**Comment by bwiernik** [comment]
Actually, the more that I think about it this is an issue even for BibTeX. If a person has stored the first letter of a title in lowercase, that is only because they are expecting Zotero to leave it like that; so, to achieve consistent formatting with BibTeX, which would otherwise leave the first letter Captitalized when converting to sentence case, this first word should be escaped (and in lowercase) when exporting to Better BibTeX.

**Comment by retorquere** [comment]
I'm still pretty ambivalent about this. Clearly the best (if least convenient) way is to add `nocase` around this word. This is the supported behavior that will do what everyone expects.

**Comment by bwiernik** [comment]
To clarify, nocase is not what would work universally. Chicago and MLA would want this capitalized, but sentence case styles wouldn’t. not uppercasing the first letter (no braces, sorry for the confusion) would yield correct results across the board and consistent behavior with Zotero. 
		
		

		
	




On Wed, Jun 20, 2018 at 9:05 PM +0200, "Emiliano Heyns" <notifications@github.com> wrote:












I'm still pretty ambivalent about this. Clearly the best (if least convenient) way is to add nocase around this word. This is the supported behavior that will do what everyone expects.

**Comment by retorquere** [comment]
I would say rather that Chicago and MLA *do* capitalize this -- in Zotero. Whether it *should* be capitalized is a different matter (or we wouldn't need `nocase` at all).

The MWE below for example does *not* capitalize. So the behavior between Zotero and BibTeX is not cleanly mappable; `psychmeta` cannot be protected at time of export because that prevents case-meddling as citeproc clearly does with MLA, but at the same time Bib(La)TeX does no case-meddling here. Whereas `<span class="nocase">psychmeta</span>: Psychometric Meta-Analysis Toolkit` deterministically gets you a lowercase psychmeta everywhere, regardless of style, both in Zotero and in Bib(La)TeX.

```
\documentclass{article}

\usepackage[american]{babel}
\usepackage[style=mla]{biblatex}

\usepackage{filecontents}
\begin{filecontents*}{test.bib}
@unpublished{Dahlkepsychmetapsychometricmetaanalysis2018,
  author = {Dahlke, Jeffrey A. and Wiernik, Brenton M.},
  shorttitle = {psychmeta},
  title = {psychmeta: Psychometric Meta-Analysis Toolkit},
  date = {2018-06}
}
\end{filecontents*}
\addbibresource{test.bib}
\begin{document}

From \cite{Dahlkepsychmetapsychometricmetaanalysis2018}

\printbibliography

\end{document}
```

**Comment by retorquere** [comment]
7189 does not titlecase the first letter if it is lowercase. I'm still not sure whether this is the right thing to do, but you can test to see if it it least does what you want.

**Comment by retorquere** [comment]
@bwiernik I could really use feedback on 7189

**Comment by bwiernik** [comment]
Sorry. Had a major deadline. Will try to test today.

**Comment by bwiernik** [comment]
No, I'm not able to get consistent results that yield the correct formatting across styles, so let's just leave the behavior as it is current. Sorry for the hassle.
